React/src/app/utils/check-board.ts

```diff
@@ -14,44 +12,8 @@ const WIN_LINES = [
     [[0, 2], [1, 1], [2, 0]],
 ];
 
-export function checkBoard(board: string[][]): WinResult | void {
-    const p1s: number[][] = [];
-    const p2s: number[][] = [];
+export function checkBoard(board: string[][], currentPlayer: string): number[][] | void {
 
-    board.forEach((row, i) => {
-        row.forEach((col, j) => {
-            if (col === P1) {
-                p1s.push([i, j]);
-            } else if (col === P2) {
-                p2s.push([i, j]);
-            }
-        });
-    });
+    return WIN_LINES.find(WIN_LINE => WIN_LINE.every(e => board[e[0]][e[1]] == currentPlayer))
```

Comment:
please choose better names for the `WIN_LINE` and `e` local variables.

Reply:
sure